docs/plugins/npe2_getting_started.md

```diff
@@ -0,0 +1,286 @@
+`setup.py` file using the _[entry point group][epg]_: `napari.manifest`. For
+example, this would be the section for a plugin `npe2-tester` with
+`napari.yaml` as the manifest file:
+
+```cfg
+[options.entry_points]
+napari.manifest =
+    npe2-tester = npe2_tester:napari.yaml
+```
+
+The manifest file is specified relative to the submodule root path.
+So for the example it will be loaded from: `<path/to/npe2-tester>/napari.yaml`.
+
+The manifest file also needs to be included as _[package data][pd]_ in
+distributable forms for the package. For example:
+
+```toml
+[metadata]
+...
+include_package_data=True
+
+[options.package_data]
+npe2_tester =
+    napari.yaml
+```
+
+A user can install napari and your plugin with:
+
+```bash
+pip install napari myplugin
+```
+
+## 4. Preparing for release
+
+Use the `Framework :: napari` [classifier](https://pypi.org/classifiers/) in
+your package's core metadata to make your plugin more discoverable. If you used
+the cookiecutter, this has already been done for you.
+
+Once your package, with its `Framework :: napari` classifier, is listed on
+PyPI, it will also be visible on the [napari hub][hub], alongside all other
+napari plugins.
+
+You can customize your plugin’s listing for the hub by following this
+[guide][hubguide].
+
+The napari hub reads the metadata of your package and displays it in a number
+of places so that users can easily find your plugin and decide if it provides
+functionality they need. Most of this metadata lives inside your package
+configuration files.
+
+You can also include a napari hub specific description file at
+/.napari/DESCRIPTION.md. The hub preferentially displays this file over your
+repository’s README.md when it’s available. This file allows you to maintain a
+more developer/repository oriented README.md while still making sure potential
+users get all the information they need to get started with your plugin.
+
+Finally, once you have curated your package metadata and description, you can
+preview your metadata, and check any missing fields using the `napari-hub-cli`
+tool. Install this tool using
+
+```bash
+pip install napari-hub-cli
+```
+
+and preview your metadata with
+
+```bash
+napari-hub-cli preview-metadata ./my-npy-reader
+```
+
+For more information on the tool see the repository README](https://github.com/chanzuckerberg/napari-hub-cli).
+
+If you want your plugin to be available on PyPI, but not visible on the napari hub,
+submit an issue on the [napari hub repository]
+(https://github.com/chanzuckerberg/napari-hub/issues/new) or send an email to
+`team@napari-hub.org` and it will be removed.
+
+## 5. Share your plugin with the world
+
+Once you are ready to share your plugin, [upload the Python package to
+PyPI][pypi-upload] and it can then be installed with a simple `pip install mypackage`. If you used the {ref}`plugin-cookiecutter-template`, you can also
+[setup automated deployments][autodeploy].
+
+If you are using Github, add the ["napari-plugin"
```

Comment:
I wonder if we should just remove this? This is definitely leftover from "pre-napari-framework" times I think. @sofroniewn ?

Reply:
I'm for leaving this as for now, and revising the hub related docs as a separate issue. Especially since it my depend on release of other features on the hub side.